scrapy/utils/misc.py

```diff
@@ -116,4 +116,28 @@ def md5sum(file):
 def rel_has_nofollow(rel):
     """Return True if link rel attribute has nofollow type"""
     return True if rel is not None and 'nofollow' in rel.split() else False
-    
+
+def construct_from_settings_or_crawler(objcls, settings, crawler, *args,
```

Comment:
@jdemaeyer Well that's a very specific case that must happen before the crawler, on current version; but I wouldn't say its not. 
I like your idea, but I will always trigger the warning when `from_settings` is present.